### PR TITLE
Internationalisation: Mark up `@grafana/ui` for translations

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- fixes remaining `no-untranslated-strings` violations in `@grafana/ui`

**Why do we need this feature?**

- more stuff translated!

**Who is this feature for?**

- anyone using other languages in grafana

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/DataLinks/DataLinksInlineEditor/DataLinksListItem.tsx [hunk 1]
```diff
@@ -51,7 +52,9 @@ export const DataLinksListItem = ({ link, onEdit, onRemove, index, itemKey }: Da
               </div>
               {isCompactExploreUrl && (
                 <FieldValidationMessage>
-                  Explore data link may not work in the future. Please edit.
+                  <Trans i18nKey="grafana-ui.data-links-list-item.field-validation">
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/DataLinks/DataLinksInlineEditor/DataLinksListItem.tsx:55]: I think the `field-validation` isn't quite describing the content here

File: packages/grafana-ui/src/components/DataLinks/FieldLinkList.tsx [hunk 2]
```diff
@@ -30,7 +31,9 @@ export function FieldLinkList({ links }: Props) {
         return <DataLinkButton key={i} link={link} />;
       })}
       <div className={styles.wrapper}>
-        <p className={styles.externalLinksHeading}>External links</p>
+        <p className={styles.externalLinksHeading}>
+          <Trans i18nKey="grafana-ui.field-link-list.external-links-heading">External links</Trans>
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/DataLinks/FieldLinkList.tsx:35]: Should a paragraph be named as a heading?

**Review by tskarhed — COMMENTED** [review]
Just some nits

File: packages/grafana-ui/src/components/DataLinks/FieldLinkList.tsx (hunk 2 above)
Comment (ashharrison90) [reply packages/grafana-ui/src/components/DataLinks/FieldLinkList.tsx:35]: i think if anything it's _more_ correct to call it a heading, and the actual problem is using a `<p>` element for (essentially) spacing reasons. but i'm definitely not fixing that as part of this PR 😅

**Review by eledobleefe — APPROVED** [review]
This is great! 🎉

**Pull request merged**